### PR TITLE
Support passing the `content_length` information when generating a URL to upload a multi-part part

### PR DESCRIPTION
**Pull request by pepicrft**
### Short description 📝
As the title states, the API supports now passing the `content_length` (optional) parameter, which the server uses to generate a pre-signed URL to upload a part. That way the storage can validate if the size of the part that has been uploaded matches the one that was originally indicated when generating the pre-signed URL.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistServer/Services/MultipartUploadGenerateURLCacheService.swift [hunk 1]
```diff
@@ -52,11 +53,13 @@ public final class MultipartUploadGenerateURLCacheService: MultipartUploadGenera
         name: String,
         cacheCategory: RemoteCacheCategory,
         uploadId: String,
-        partNumber: Int
+        partNumber: Int,
+        contentLength: Int?
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply Sources/TuistServer/Services/MultipartUploadGenerateURLCacheService.swift]: The `contentLength` option is used. I updated it to make it `required` to force all the business logic in the code base to always pass it.

File: Sources/TuistServer/Services/MultipartUploadArtifactService.swift [hunk 2]
```diff
@@ -41,7 +41,7 @@ enum MultipartUploadArtifactServiceError: FatalError {
 public protocol MultipartUploadArtifactServicing {
     func multipartUploadArtifact(
         artifactPath: AbsolutePath,
-        generateUploadURL: @escaping (Int) async throws -> String
+        generateUploadURL: @escaping ((partNumber: Int, contentLength: Int)) async throws -> String
```
Comment (fortmarek) [inline Sources/TuistServer/Services/MultipartUploadArtifactService.swift:44]: I'd consider changing the tuple to be a struct, such as `MultipartUploadPart`.

**Review by fortmarek — COMMENTED** [review]
Should some of the changes here be reflected in unit tests? 🤔

**Comment by pepicrft** [comment]
> Should some of the changes here be reflected in unit tests? 🤔

The call-back approach of the implementation makes it tricky to test it because we'd need to capture the callback being passed to the mock, and then invoke it manually. We should be able to do it with [perform](https://kolos65.github.io/Mockable/documentation/mockable/functionactionbuilder/perform(_:)/), but I'll have to update Mockable first.

**Comment by pepicrft** [comment]
I'll improve the tests once we [merge this](https://github.com/tuist/tuist/pull/6884)

**Comment by pepicrft** [comment]
@fortmarek I added tests here that ensure the callbacks do the right job of calling the service that generates the URL.

**Review by fortmarek — APPROVED** [review]
Thanks for updating the tests!

**Pull request merged**